File: util/layout.c
#include <stdio.h>

typedef struct Nested {
  struct {
    long big;
    char small;
  } inner;
  char last;
} Nested;

typedef struct Together {
  long big;
  char small;
  char last;
} Together;

int main() {
  {
    Nested value;
    printf("Nested: \n");
    printf("big: %ld\n", (long)&value.inner.big - (long)&value);
    printf("small: %ld\n", (long)&value.inner.small - (long)&value);
    printf("last: %ld\n", (long)&value.last - (long)&value);
  }
  {

    Together value;
    printf("---\nTogether: \n");
    printf("big: %ld\n", (long)&value.big - (long)&value);
    printf("small: %ld\n", (long)&value.small - (long)&value);
    printf("last: %ld\n", (long)&value.last - (long)&value);
  }
  return 0;
}

File: util/layout.zig
const Inner = struct  {
    big: i64,
    small: u8,
};

const Nested = struct {
    inner: Inner,
    last: u8,
};

const Together = struct {
    big: i64,
    small: u8,
    last: u8,
};

const print = @import("std").debug.print;
pub fn main() void {
    {
        var value: Nested = undefined;
        print("Nested: \n", .{});
        const start = @intFromPtr(&value);
        print("big: {}\n", .{ (@intFromPtr(&value.inner.big) ) - start});
        print("small: {}\n", .{ (@intFromPtr(&value.inner.small) ) - start});
        print("last: {}\n", .{ (@intFromPtr(&value.last) ) - start});
    }
    {
        var value: Together = undefined;
        print("Together: \n", .{});
        const start = @intFromPtr(&value);
        print("big: {}\n", .{ (@intFromPtr(&value.big) ) - start});
        print("small: {}\n", .{ (@intFromPtr(&value.small) ) - start});
        print("last: {}\n", .{ (@intFromPtr(&value.last) ) - start});
    }
}
